_layouts/single.html

```diff
@@ -17,21 +17,30 @@
 <div id="main" role="main">
   {% include sidebar.html %}
 
-  <article class="page" itemscope itemtype="https://schema.org/CreativeWork">
+  <article class="page h-entry" itemscope itemtype="https://schema.org/CreativeWork">
     {% if page.title %}<meta itemprop="headline" content="{{ page.title | markdownify | strip_html | strip_newlines | escape_once }}">{% endif %}
     {% if page.excerpt %}<meta itemprop="description" content="{{ page.excerpt | markdownify | strip_html | strip_newlines | escape_once }}">{% endif %}
     {% if page.date %}<meta itemprop="datePublished" content="{{ page.date | date_to_xmlschema }}">{% endif %}
     {% if page.last_modified_at %}<meta itemprop="dateModified" content="{{ page.last_modified_at | date_to_xmlschema }}">{% endif %}
 
+  <ul class="hidden">
```

Comment:
Is there a better way of applying microformats for this data? This is the sort of extraneous markup I wanted to avoid, especially since it's duplicating the published/modified datetimes from above along with the authorship that is already being applied the author sidebar.

I get all sorts of "code smells" from it not being DRY and having to hide it all via CSS.

Reply:
Yes—the repeated variables bother me, too.   The driving cause for that is how microformats need the data to be the contents of the enclosing tag and Schema.org puts the data in attributes.  Specifically:
> microformats do not use `link` or `meta` elements within the content of the page and in some cases require particular elements to be used to encode information, such as using `abbr` to support the datetime-design-pattern as illustrated by the `dtstart` property in the example above.  <cite>("<a href="https://www.w3.org/wiki/Mixing_HTML_Data_Formats#Mixing_Syntaxes">Mixing Syntaxes</a>" heading at W3C's "Mixing HTML Data Formats" page)<cite>

I used existing tags where possible (the `p-name` class for the <h1> article title and the `e-content` class in the article <section> tag).  The contents of the hidden <ul>, though, were the values from the Schema.org <meta> tags.

For a while, I was putting microformat markup in the `_includes/page__meta.html`, but separating the article microformat data across two template files felt like a worse compromise as compared to keeping all of the HTML data formats in close proximity in one template file.

Reply:
With all of the "hidden" list element items, which are actually required? Can we move forward with the other data you marked up that isn't hidden from view?

I'm still iffy on the duplicating all the author markup as it's in the sidebar with microformat markup now.

Can the published/modified dates be marked up in [`_includes/page_date.html`](https://github.com/mmistakes/minimal-mistakes/blob/master/_includes/page__date.html) instead? Neither of those are hidden and you wouldn't have to duplicate the markup then.